### ISSUE TITLE
Query on isarLink not working as expected

### ISSUE THREAD

**Issue by gabzim** [isar/isar#390]
**Steps to Reproduce**
I can't get to work queries on `IsarLinks`. I have a list of quotes, each quotes has an author, I'm trying to get all quotes written by a given author. I'm not sure I can get it to work or whether I understand the query language correctly. BackLinks work, but querying on quotes does not.
**Code sample**
```
@Collection()
class Author {
  @Id()
  int? id;
  @Index()
  late String name;
  String? summary;
  String? reference;
  @Backlink(to: "author")
  IsarLinks<Quote> quotes = IsarLinks<Quote>();
}

@Collection()
class Quote {
  int? id;
  late String content;
  late String source;
  int? year;
  final author = IsarLink<Author>();
}
```
When I do a query such as 
```
getQuotesByAuthor(int authorId) => isar.quotes.filter().author((q) => q.idEqualTo(authorId)).findAll();
```  
 Nothing comes back. I'm not sure I fully understand how to query IsarLinks. If I try to load the author and then do `await author.quotes.load();` That'll work just fine and return all the quotes for that given author. Futhermore, any kind of query returns empty results, such as: 
```
isar.quotes.filter().author((q) => q.idGreaterThan(0)).findAll();
```
Am I misunderstanding the point of querying isarlinks? 
**Version**
 - Platform: iOS
 - Flutter version: 3.0.0
 - Isar version: 2.5.0
 - Isar Generator version: 2.5.0

**Comment by md-weber** [comment]
Possible Dup: https://github.com/isar/isar/issues/370

**Comment by leisim** [comment]
Fixed in `3.0.0-dev.8`
